File: server/node/create-list.js
import Courier from '@trycourier/courier';
import dotenv from 'dotenv';

dotenv.config({ path: new URL('../.env', import.meta.url) });

const apiKey = process.env.COURIER_API_KEY;
const listId = process.env.COURIER_CREATE_LIST_LIST_ID;
const listName = process.env.COURIER_CREATE_LIST_LIST_NAME || 'My List Name';

const client = new Courier({ apiKey });

await client.lists.update(listId, {
  name: listName,
  preferences: {
    categories: {},
    notifications: {}
  }
});

File: server/python/create-list.py
import os
from pathlib import Path
from dotenv import load_dotenv
from courier import Courier

load_dotenv(Path(__file__).parent.parent / ".env")

api_key = os.getenv("COURIER_API_KEY")
list_id = os.getenv("COURIER_CREATE_LIST_LIST_ID")
list_name = os.getenv("COURIER_CREATE_LIST_LIST_NAME", "My List Name")

client = Courier(api_key=api_key)

client.lists.update(
    list_id=list_id,
    name=list_name,
    preferences={
        "categories": {},
        "notifications": {}
    }
)

